### PR TITLE
Safe API for instruments

### PR DESCRIPTION
**Pull request by runarorama**

### CODE REVIEW

**Comment by runarorama** [comment]
Tests are hanging for some reason.

File: core/build.sbt [hunk 1]
```diff
@@ -14,9 +14,12 @@ libraryDependencies ++= Seq(
   "com.twitter"       %% "algebird-core"        % "0.8.0",
   "org.fusesource"     % "sigar"                % "1.6.4" classifier("native") classifier("") exclude("log4j", "log4j"),
   "org.slf4j"          % "log4j-over-slf4j"     % "1.7.+", // SIGAR requires the log4j legacy API
-  "oncue.journal"     %% "core"                 % "2.2.1"
+  "oncue.journal"     %% "core"                 % "2.2.1",
+  "org.spire-math"    %% "spire"                % "0.11.0"
```
Comment (timperrett) [inline core/build.sbt]: What are we using Spire for specifcally? It's a pretty big dependency and we already have algebird. I'd be fine to use one of the other, but both seems a bit silly, no?

Comment (runarorama) [inline core/build.sbt]: Specifically for the `Natural` type. It's a crime that this type isn't in the standard library, but here we are. This is in an effort to make the `Counter` API more safe.

Comment (runarorama) [inline core/build.sbt]: Incidentally, I just tried taking out Algebird and replacing it entirely with Spire. But we're using the `Moments` type from Algebird which Spire doesn't have.

**Comment by runarorama** [comment]
Some of `MonitoringSpec` is still hanging for some reason. Giving up for now and getting back to it later.

**Pull request closed without merging**